### PR TITLE
Improvements/known changes

### PR DESCRIPTION
**Pull request by Josephfallen**
Please approve it

### CODE REVIEW

File: src/lib/elements/NavBar.svelte [hunk 1]
```diff
@@ -4,16 +4,29 @@
...
         text-decoration: none;
+        padding: 0.5rem 1rem;
+        transition: color 0.3s ease, background-color 0.3s ease, transform 0.3s ease;
+        border-radius: 4px;
     }
     nav > a:hover {
-        color: #AAA;
+        color: #fff;
+        background-color: #555;
+        transform: scale(1.1);
```
Comment (TAG-Epic) [inline src/lib/elements/NavBar.svelte]: This will cause layout to shift?

File: src/routes/+page.svelte [hunk 2]
```diff
@@ -21,53 +17,118 @@
 
     async function onSearchFieldChange(event: InputEvent) {
         if (searchQuery.trim() === "") {
-            $searchResults = [];
+            searchResults.set([]);
```
Comment (TAG-Epic) [inline src/routes/+page.svelte]: Why was this changed?

File: src/routes/+page.svelte [hunk 3]
```diff
@@ -21,53 +17,118 @@
...
-    .user:hover {
-        color: #555;
-    }
-
-</style>
-
 <svelte:head>
     <title>Vail stats far from this world</title>
     <meta name="description" content="View official vail stats for any vail user">
+    <style>
```
Comment (TAG-Epic) [inline src/routes/+page.svelte]: Prefer local css (aka not in <svelte:head>)
And use sveltes global css feature instead of this

File: src/routes/+page.svelte [hunk 4]
```diff
@@ -21,53 +17,118 @@
...
+
+    <h1>Search for user</h1>
+    <form on:submit|preventDefault={() => {}}>
+        <label for="username-field">Username</label>
+        <input id="username-field" type="search" placeholder={userPlaceholder} on:input={debounce(onSearchFieldChange)} bind:value={searchQuery}>
+    </form>
+
+    <div class="users">
+        {#each $searchResults as user}
+            <button class="user" on:click={() => {goto(`/players/${user.id}`)}}>
```
Comment (TAG-Epic) [inline src/routes/+page.svelte]: I wanted this to be a link but couldn't figure it out. Could you try?

File: src/routes/.internal/open-graph/player-stats/[userId].png/+server.ts [hunk 5]
```diff
@@ -1,44 +1,43 @@
...
+        // Filter out guns without shot data and create a combination virtual gun
         const guns = Object.values(userStats.weapons).filter(weapon => weapon.shots !== undefined);
         const gun = createCombinationVirtualGun(guns);
-        const gunStore = writable<UserGunStats>(gun);
-        const renderResult = ShotVisualizer.render({svg: true, stats: gunStore});
-        svg = renderResult.html;
+        const gunStore = writable(gun);
+        
+        // Render ShotVisualizer Svelte component with SVG output
+        const { html } = ShotVisualizer.render({ svg: true, stats: gunStore });
```
Comment (TAG-Epic) [inline src/routes/.internal/open-graph/player-stats/[userId].png/+server.ts]: Why was this moved in the try?

File: src/routes/players/[userId]/+page.svelte [hunk 6]
```diff
@@ -1,3 +1,4 @@
+<!-- Shot CSS file-->
```
Comment (TAG-Epic) [inline src/routes/players/[userId]/+page.svelte]: Remove

File: src/routes/players/[userId]/+page.svelte [hunk 7]
```diff
@@ -24,33 +25,111 @@
...
+            {#each Object.keys(stats.weapons) as weaponId}
+                <option value={weaponId}>{weaponId}</option>
+            {/each}
+        </select>
+        <h3>Hits</h3>
+        <ShotVisualizer stats={activeWeaponStats} />
+        <!-- Placeholder for the pie chart -->
+        <svg class="pie-chart" viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg">
+            <!-- Draw the pie chart here -->
+        </svg>
```
Comment (TAG-Epic) [inline src/routes/players/[userId]/+page.svelte]: Move to seperate component?

**Review by TAG-Epic — CHANGES_REQUESTED** [review]

File: src/lib/elements/NavBar.svelte (hunk 1 above)
Comment (Josephfallen) [reply src/lib/elements/NavBar.svelte]: It will look like how the Screenshots have looked

File: src/routes/players/[userId]/+page.svelte (hunk 7 above)
Comment (Josephfallen) [reply src/routes/players/[userId]/+page.svelte]: In the future just there as a place holder

File: src/routes/players/[userId]/+page.svelte (hunk 6 above)
Comment (Josephfallen) [reply src/routes/players/[userId]/+page.svelte]: Have it there so I can figure out which one is which

File: src/routes/.internal/open-graph/player-stats/[userId].png/+server.ts (hunk 5 above)
Comment (Josephfallen) [reply src/routes/.internal/open-graph/player-stats/[userId].png/+server.ts]: It recentered it irc
 I used chat gpt for this one

File: src/routes/+page.svelte (hunk 4 above)
Comment (Josephfallen) [reply src/routes/+page.svelte]: I will try it, What should it look like?

File: src/routes/+page.svelte (hunk 3 above)
Comment (Josephfallen) [reply src/routes/+page.svelte]: This is what a buddy recommended, as I do not have experience with svelt

File: src/routes/+page.svelte (hunk 2 above)
Comment (Josephfallen) [reply src/routes/+page.svelte]: Runs faster from what I have seen

File: src/lib/elements/NavBar.svelte (hunk 1 above)
Comment (TAG-Epic) [reply src/lib/elements/NavBar.svelte]: Yeah but when hovered it will cause other elements to move

File: src/routes/+page.svelte (hunk 2 above)
Comment (TAG-Epic) [reply src/routes/+page.svelte]: Should be negligble

File: src/lib/elements/NavBar.svelte (hunk 1 above)
Comment (Josephfallen) [reply src/lib/elements/NavBar.svelte]: > Yeah but when hovered it will cause other elements to move

It should not from my tests

File: src/routes/+page.svelte (hunk 3 above)
Comment (TAG-Epic) [reply src/routes/+page.svelte]: Right don't do that. Would recommend using the sveltekit tutorial/docs as they know how shit is done

File: src/routes/+page.svelte (hunk 2 above)
Comment (Josephfallen) [reply src/routes/+page.svelte]: > Should be negligble

Alright I will change it back then

File: src/routes/+page.svelte (hunk 4 above)
Comment (TAG-Epic) [reply src/routes/+page.svelte]: The same. Want to change it so svelte optimises things

File: src/routes/+page.svelte (hunk 3 above)
Comment (Josephfallen) [reply src/routes/+page.svelte]: image
This is what happened when changing to local and global css

File: src/routes/+page.svelte (hunk 4 above)
Comment (Josephfallen) [reply src/routes/+page.svelte]: So make it not have the user ID in the link?

File: src/routes/+page.svelte (hunk 3 above)
Comment (TAG-Epic) [reply src/routes/+page.svelte]: Then you did it wrong. Send me what you did on discord

Comment (Josephfallen) [reply src/routes/+page.svelte]: > Then you did it wrong. Send me what you did on discord

I'm going to mess with it a tad more will update when I do
